modules/code-builder/docs/PRD.md

```diff
@@ -0,0 +1,177 @@
+   - Play Note Statement Brick
+   - Rest Statement Brick
+   - Set Instrument Statement Brick
+   - Increase Volume Statement Brick
+   - Decrease Volume Statement Brick
+   - Set Tempo Statement Brick
+   - Set Key Statement Brick
+   - Set Mode Statement Brick
+   - Start Drum Statement Brick
+   - Start Chord Statement Brick
+
+     ![example](./images/image-3.png)
+
+#### 4. Brick Bricks:
+   - Repeat Brick
+   - Forever Brick
+   - If Brick
+   - If-Else Brick
+   - Action Brick
+   - Start Brick
+   - Clamp Brick
+   - Store In Brick
+   - Box Brick
+   - Pitch Brick
+   - Rhythm Brick
+   - Meter Brick (representing time signatures)
+   - Volume Brick
+   - Articulation Brick
+   - Pitch-Time Matrix Brick
+
+### b. Brick Appearance
+   - **Distinct Shapes**: Each brick type has a unique shape to differentiate its function visually.
+   - **Colors**: Color-coding helps users quickly identify the type of brick.
+   - **Icons and Labels**: Visual symbols and text labels indicate the brick's function.
+   - **Input/Output Ports**: Connectors visually represent where bricks can attach to each other.
+   - **Editable Text Labels/Fields**: Users can input data directly into the bricks, such as note names, durations, and numerical values.
+
+### c. Brick Interactions
+   - **Inline Text Editing**: Users can click on text fields within bricks to edit labels and values directly.
+   ![alt text](./images/image-4.png)
+   - **Drop-Down Menus/Selectors**: For choosing predefined options, enhancing ease of use.
+   - **Dedicated Editors/Dialogs**: For more complex properties, a separate interface allows detailed configuration.
+   ![alt text](./images/image-5.png)
+   - **Resizing and Rotation**: Bricks can be resized and rotated to fit the workspace better.
+   - **Cloning/Duplication**: Users can easily create copies of bricks for repeated use.
+   - **Context Menus**: Right-click options for advanced customization and settings.
+
+## 2. Stack of Bricks
+
+### a. Stack Validation:
+   - **Visual Feedback**: Indicators show whether brick combinations are valid.
+
+      ![alt text](./images/image-6.png)
+   - **Error Indicators**: Explanations for incompatible connections help users troubleshoot.
+        - Add a reddish boundary for users to easily tell whether the bricks are mergeable or not.  
+   ![alt text](./images/image-7.png) 
+
+
+   - **Disable Validation**: Temporarily turn off validation for complex or experimental setups.
+
+### b. Stack Editing:
+   - **Connection Editing**: Options for rerouting, splitting, or merging connections.
+   - **Quick Edit Shortcuts**: Context menus or keyboard shortcuts speed up editing.
+   - **Undo/Redo**: Users can revert or reapply changes to their stacks.
+
+### c. Stack Grouping:
+   - **Grouping Bricks**: Multiple bricks or connections can be grouped for better organization.
+   ![alt text](./images/image-8.png)
+   - **Visual Boundaries**: Outlines or boundaries define groups clearly.
```

Comment:
- **Grouping Bricks**: Multiple bricks or connections can be grouped together to form a single unit, allowing for better organization and easier management. For example, if you have a set of bricks that define a musical phrase, you can group them together to move or edit them as one.

     
![image](https://github.com/sugarlabs/musicblocks-v4/assets/143683619/bb58bc7f-2faf-4fd6-a5aa-71f275b352e6)


   - **Visual Boundaries**: Clear outlines or boundaries around groups of bricks visually indicate the grouped unit, helping users distinguish between different groups. This might involve a colored border or a shaded background around the group to make it stand out.  (This is an additional feature to be discussed with community)